### PR TITLE
(port) adds contorionist jumpsuit

### PR DESCRIPTION
**Pull request by boodaliboo**
https://github.com/ParadiseSS13/Paradise/pull/4304 original pr
adds a new atmos tech exclusive jumpsuit that allows them to crawl through vents. it looks like a atmostech jumpsuit so its stealthy
#### Changelog

:cl:  
rscadd: Added new antag item
/:cl:

### CODE REVIEW

**Comment by boodaliboo** [comment]
motherfucker

File: code/modules/uplink/uplink_items.dm [hunk 1]
```diff
@@ -1812,6 +1812,13 @@ GLOBAL_LIST_INIT(uplink_items, subtypesof(/datum/uplink_item))
 	cost = 4
 	surplus = 0
 	restricted_roles = list("Cook", "Botanist", "Clown", "Mime")
+	
+/datum/uplink_item/jobspecific/contortionist
```
Comment (missatessatessy) [inline code/modules/uplink/uplink_items.dm]: /datum/uplink_item/role_restricted/contortionist

File: code/modules/uplink/uplink_items.dm [hunk 2]
```diff
@@ -1812,6 +1812,13 @@ GLOBAL_LIST_INIT(uplink_items, subtypesof(/datum/uplink_item))
...
+	name = "Contortionist's Jumpsuit"
+	desc = "A highly flexible jumpsuit that will help you navigate the ventilation loops of the station internally. Comes with pockets and ID slot, but can't be used without stripping off most gear, including backpack, belt, helmet, and exosuit. Free hands are also necessary to crawl around inside."
+	item = /obj/item/clothing/under/contortionist
+	cost = 6
+	job = list("Atmospheric Technician")
```
Comment (missatessatessy) [inline code/modules/uplink/uplink_items.dm]: restricted_roles =

File: code/modules/clothing/under/miscellaneous.dm [hunk 3]
```diff
@@ -867,3 +867,30 @@
...
+	..()
+
+/obj/item/clothing/under/contortionist/dropped(mob/living/carbon/human/user)
+	if(!user.get_int_organ(/obj/item/organ/internal/heart/gland/ventcrawling))
+		user.ventcrawler = 0
+	..()
+
+/obj/item/clothing/under/contortionist/proc/check_clothing(mob/user as mob)
+	//Allowed to wear: glasses, shoes, gloves, pockets, mask, and jumpsuit (obviously)
+	var/list/slot_must_be_empty = list(slot_back,slot_handcuffed,slot_legcuffed,slot_l_hand,slot_r_hand,slot_belt,slot_head,slot_wear_suit)
```
Comment (missatessatessy) [inline code/modules/clothing/under/miscellaneous.dm:891]: back, handcuffed, legcuffed, l_store, r_store, head, wear_suit, belt
see code/modules/mob/living/carbon/human/inventory.dm for why

File: code/modules/clothing/under/miscellaneous.dm [hunk 4]
```diff
@@ -867,3 +867,30 @@
...
+	item_color = "atmos"
+	resistance_flags = FIRE_PROOF
+
+/obj/item/clothing/under/contortionist/equipped(mob/living/carbon/human/user, slot)
+	if(!user.ventcrawler)
+		user.ventcrawler = 2
+	..()
+
+/obj/item/clothing/under/contortionist/dropped(mob/living/carbon/human/user)
+	if(!user.get_int_organ(/obj/item/organ/internal/heart/gland/ventcrawling))
```
Comment (missatessatessy) [inline code/modules/clothing/under/miscellaneous.dm:885]: proc, nor organ exist

**Comment by boodaliboo** [comment]
ventcrawler = VENTCRAWLER_ALWAYS

**Pull request closed without merging**